src/emotion/Button.tsx

```diff
@@ -0,0 +1,62 @@
+/* @jsx jsx */
+
+import { jsx } from '@emotion/core';
+import { FunctionComponent, HTMLAttributes } from 'react';
+
+import { Box, IBox } from './Box';
+import { useTheme } from './theme';
+
+export const Button: FunctionComponent<IButton> = props => {
+  const { as = 'button', disabled, ...rest } = props;
+
+  const css = [...buttonStyles({ disabled })];
+
+  return jsx(Box, {
+    ...rest,
+    as,
+    css,
+  });
+};
+
+export interface IButton extends IButtonProps, IBox, HTMLAttributes<'button'> {}
```

Comment:
Let's have a generics here - 'button' should be a default, but not hardcoded.
Why? `As` is not enforced and can be changed.

Reply:
Looking at signatures, I think we have to use HTMLButtonElement here. It's primarily used in SyntheticEvent, so I think we can live with button here.

Reply:
Agreed on both! And as a general pattern in all the components (Box, Flex, etc).